### PR TITLE
issue-1482: Minor package refactoring and cleanup

### LINKED ISSUE #1482: Minor package refactoring and cleanup under the strongbox-maven-metadata-api module
# Task Description

The [`strongbox-maven-metadata-api`](https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api) module requires some minor package refactoring.

# Tasks

The following tasks will need to be carried out:
* [ ] Create a new package called `org.carlspring.strongbox.storage.metadata.maven` under the [`strongbox-maven-metadata-api`](https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api) module.
* [ ] Move all the sub-packages and direct classes in [`strongbox-maven-metadata-api`](https://github.com/strongbox/strongbox/tree/master/strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api) from `org.carlspring.strongbox.storage.metadata` to `org.carlspring.strongbox.storage.metadata.maven`.
* [ ] Move the `org.carlspring.strongbox.io.filters.ArtifactVersionDirectoryFilter` under `org.carlspring.strongbox.maven.io.filters.ArtifactVersionDirectoryFilter`.
* [ ] Make sure this is applied to to the test code.

# Help

* [Our chat](https://chat.carlspring.org/)
* Points of contact:
  * @carlspring
  * @fuss86
  * @sbespalov
  * @steve-todorov

### PR DESCRIPTION
**Pull request by leonmak**
Under the strongbox-maven-metadata-api module

# Pull Request Description

This pull request closes #1482 .

# Acceptance Test

* [x] Building the code with `mvn clean install -Dintegration.tests` still works.
* [x] Running `mvn spring-boot:run` in the `strongbox-web-core` still starts up the application correctly.
* [x] Building the code and running the `strongbox-distribution` from a `zip` or `tar.gz` still works.
* [x] The tests in the [`strongbox-web-integration-tests`](https://github.com/strongbox/strongbox-web-integration-tests/) still run properly.

# Questions

* Does this pull request break backward compatibility? 
  * [ ] Yes
  * [x] No

* Does this pull request require other pull requests to be merged first? 
  * [ ] Yes, please see #...
  * [x] No

* Does this require an update of the documentation?
  * [ ] Yes, please see strongbox/strongbox-docs#{PR_NUMBER}
  * [x] No

### CODE REVIEW

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/VersionCollector.java [hunk 1]
```diff
@@ -1,10 +1,10 @@
 package org.carlspring.strongbox.storage.metadata;
 
-import org.carlspring.strongbox.io.filters.ArtifactVersionDirectoryFilter;
-import org.carlspring.strongbox.storage.metadata.comparators.MetadataVersionComparator;
-import org.carlspring.strongbox.storage.metadata.comparators.SnapshotVersionComparator;
-import org.carlspring.strongbox.storage.metadata.versions.MetadataVersion;
-import org.carlspring.strongbox.storage.metadata.visitors.ArtifactVersionDirectoryVisitor;
+import org.carlspring.strongbox.storage.metadata.maven.ArtifactVersionDirectoryFilter;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/VersionCollector.java]: Hi!
This is not what I had in mind. The sub-packages need to be moved (preserved) as well.

Comment (leonmak) [reply strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/VersionCollector.java]: Oh right! Read it wrongly, changed it

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/VersionCollector.java [hunk 2]
```diff
@@ -1,10 +1,10 @@
 package org.carlspring.strongbox.storage.metadata;
 
-import org.carlspring.strongbox.io.filters.ArtifactVersionDirectoryFilter;
-import org.carlspring.strongbox.storage.metadata.comparators.MetadataVersionComparator;
-import org.carlspring.strongbox.storage.metadata.comparators.SnapshotVersionComparator;
-import org.carlspring.strongbox.storage.metadata.versions.MetadataVersion;
-import org.carlspring.strongbox.storage.metadata.visitors.ArtifactVersionDirectoryVisitor;
+import org.carlspring.strongbox.storage.metadata.maven.filters.ArtifactVersionDirectoryFilter;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/VersionCollector.java]: This should be:
```
import org.carlspring.strongbox.storage.metadata.maven.io.filters.ArtifactVersionDirectoryFilter;
```

File: strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/maven/filters/ArtifactVersionDirectoryFilter.java [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package org.carlspring.strongbox.io.filters;
+package org.carlspring.strongbox.storage.metadata.maven.filters;
```
Comment (carlspring) [inline strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-maven-metadata-api/src/main/java/org/carlspring/strongbox/storage/metadata/maven/filters/ArtifactVersionDirectoryFilter.java]: This should be:
```
package org.carlspring.strongbox.storage.metadata.maven.io.filters;

```

**Comment by carlspring** [comment]
Hi!

Thank you for your contribution!

Would you mind signing the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page?

Also, please, feel free to join our [chat channel](https://chat.carlspring.org/), if you'd like to learn more about the project and/or like to find out what else you could help with.

Kind regards,

Martin

**Comment by steve-todorov** [comment]
Hi @leonmak and thanks for your contribution! :)

It looks like this is failing in Jenkins. Would you mind having another look?
If you have questions you can always join us in our [chat](https://chat.carlspring.org/).

**Comment by carlspring** [comment]
@leonmak ,

Would you mind rebasing your pull request? Thanks! :)

**Comment by carlspring** [comment]
@leonmak ,

Just a reminder -- could you please sign the ICLA, as described in the [Contributing](https://github.com/strongbox/strongbox/blob/master/CONTRIBUTING.md) page? Thanks! :)

**Comment by carlspring** [comment]
Thank you for your contrbution and congratulations on your first pull request! :)

**Pull request merged**